### DISCUSSION TITLE
maintaining topology

### DISCUSSION THREAD

**Discussion by hmalindu1** [geoman-io/leaflet-geoman#941] [Questions & Help]
Hi there, Thanks so much for your time!
I wanted to know if it is possible to have one common boundary line within two shapes. And also is it possible to maintain topology using your plugin. I have seen something like [this](https://www.youtube.com/watch?v=rWVRIoIG3gc) in leaflet.draw 
Kind reagards,
Hashan.

**Comment by Falke-Design** [comment]
Can you please explain what do you mean with `topology` and how we can support that?

We have the Pro ⭐ Feature [Pinning](https://github.com/geoman-io/leaflet-geoman#pinning-) which is your "Common Boundary". [Leaflet-Geoman Pro](https://geoman.io/leaflet-geoman-pro)

**Reply by hmalindu1** [reply]
`topology` means that when you move a vertex around, all the connected line should move along with that point, hence it makes newly adjusted polygons. Specifically within parcels their must have a common boundary. Not overlapped two lines. In the pro version their are two lines (line on top of another one). What I need to know is how to create a one common line for two polygons.

**Reply by Falke-Design** [reply]
I don't think that this is possible with Leaflet, I never saw something like this. The only possible thing is that the two lines exact below each other.

**Reply by hmalindu1** [reply]
In this question, on the given answer, They have used [topoJSON](https://github.com/topojson) using leaflet. Is that helpful? [Here](https://stackoverflow.com/questions/68111398/is-it-possible-in-leaflet-to-share-a-single-common-boundary-line-within-two-pol) I have questioned it in Stack Exchange. It will clear you what my point is.
